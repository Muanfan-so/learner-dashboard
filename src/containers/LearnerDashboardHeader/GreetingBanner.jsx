import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import { getConfig } from '@edx/frontend-platform';
import { useIntl } from '@edx/frontend-platform/i18n';
import { Image } from '@edx/paragon';

import messages from './messages';

export const GreetingBanner = ({ size }) => {
  const { formatMessage } = useIntl();

  let greetMessage;
  const hour = new Date().getHours();

  if (hour > 16) {
    greetMessage = messages.goodEvening;
  } else if (hour > 11) {
    greetMessage = messages.goodAfternoon;
  } else {
    greetMessage = messages.goodMorning;
  }

  const isSmall = size === 'small';

  return (
    <div
      className={classNames(
        'd-flex align-items-center justify-content-center',
        { 'pb-5': !isSmall, 'p-3.5': isSmall },
      )}
    >
      <a href={`${getConfig().LMS_BASE_URL}/dashboard`}>
        <Image
          style={{ width: isSmall ? '46px' : '148px' }}
          className="d-block"
          src={getConfig().LOGO_WHITE_URL}
          alt={getConfig().SITE_NAME}
        />
      </a>
      <div className={`greetings-slash-container-${size} bg-brand-500`} />
      {isSmall
        ? (
          <h5 role="presentation" className="text-center text-accent-b">
            {formatMessage(greetMessage)}
          </h5>
        ) : (
          <h1
            role="presentation"
            className="text-center text-accent-b display-1"
          >
            {formatMessage(greetMessage)}
          </h1>
        )}
    </div>
  );
};
GreetingBanner.propTypes = {
  size: PropTypes.oneOf(['small', 'large']).isRequired,
};

export default GreetingBanner;
