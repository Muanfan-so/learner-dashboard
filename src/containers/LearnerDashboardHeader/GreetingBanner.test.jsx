import { shallow } from 'enzyme';

import { GreetingBanner } from './GreetingBanner';

describe('GreetingBanner', () => {
  const morning = new Date('2021-01-01T11:59:59.999');
  const afternoon = new Date('2021-01-01T16:59:59.999');
  const evening = new Date('2021-01-01T18:00:00');
  describe('snapshots', () => {
    ['small', 'large'].forEach((size) => {
      test(`with size ${size} and morning`, () => {
        jest.useFakeTimers().setSystemTime(morning);
        const wrapper = shallow(<GreetingBanner size={size} />);
        expect(wrapper).toMatchSnapshot();
      });
      test(`with size ${size} and afternoon`, () => {
        jest.useFakeTimers().setSystemTime(afternoon);
        const wrapper = shallow(<GreetingBanner size={size} />);
        expect(wrapper).toMatchSnapshot();
      });
      test(`with size ${size} and evening`, () => {
        jest.useFakeTimers().setSystemTime(evening);
        const wrapper = shallow(<GreetingBanner size={size} />);
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
