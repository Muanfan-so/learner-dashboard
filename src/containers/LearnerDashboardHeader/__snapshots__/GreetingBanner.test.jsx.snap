// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GreetingBanner snapshots with size large and afternoon 1`] = `
<div
  className="d-flex align-items-center justify-content-center pb-5"
>
  <a
    href="http://localhost:18000/dashboard"
  >
    <Image
      alt="localhost"
      className="d-block"
      src="https://edx-cdn.org/v3/default/logo-white.svg"
      style={
        Object {
          "width": "148px",
        }
      }
    />
  </a>
  <div
    className="greetings-slash-container-large bg-brand-500"
  />
  <h1
    className="text-center text-accent-b display-1"
    role="presentation"
  >
    Good Afternoon!
  </h1>
</div>
`;

exports[`GreetingBanner snapshots with size large and evening 1`] = `
<div
  className="d-flex align-items-center justify-content-center pb-5"
>
  <a
    href="http://localhost:18000/dashboard"
  >
    <Image
      alt="localhost"
      className="d-block"
      src="https://edx-cdn.org/v3/default/logo-white.svg"
      style={
        Object {
          "width": "148px",
        }
      }
    />
  </a>
  <div
    className="greetings-slash-container-large bg-brand-500"
  />
  <h1
    className="text-center text-accent-b display-1"
    role="presentation"
  >
    Good Evening!
  </h1>
</div>
`;

exports[`GreetingBanner snapshots with size large and morning 1`] = `
<div
  className="d-flex align-items-center justify-content-center pb-5"
>
  <a
    href="http://localhost:18000/dashboard"
  >
    <Image
      alt="localhost"
      className="d-block"
      src="https://edx-cdn.org/v3/default/logo-white.svg"
      style={
        Object {
          "width": "148px",
        }
      }
    />
  </a>
  <div
    className="greetings-slash-container-large bg-brand-500"
  />
  <h1
    className="text-center text-accent-b display-1"
    role="presentation"
  >
    Good Morning!
  </h1>
</div>
`;

exports[`GreetingBanner snapshots with size small and afternoon 1`] = `
<div
  className="d-flex align-items-center justify-content-center p-3.5"
>
  <a
    href="http://localhost:18000/dashboard"
  >
    <Image
      alt="localhost"
      className="d-block"
      src="https://edx-cdn.org/v3/default/logo-white.svg"
      style={
        Object {
          "width": "46px",
        }
      }
    />
  </a>
  <div
    className="greetings-slash-container-small bg-brand-500"
  />
  <h5
    className="text-center text-accent-b"
    role="presentation"
  >
    Good Afternoon!
  </h5>
</div>
`;

exports[`GreetingBanner snapshots with size small and evening 1`] = `
<div
  className="d-flex align-items-center justify-content-center p-3.5"
>
  <a
    href="http://localhost:18000/dashboard"
  >
    <Image
      alt="localhost"
      className="d-block"
      src="https://edx-cdn.org/v3/default/logo-white.svg"
      style={
        Object {
          "width": "46px",
        }
      }
    />
  </a>
  <div
    className="greetings-slash-container-small bg-brand-500"
  />
  <h5
    className="text-center text-accent-b"
    role="presentation"
  >
    Good Evening!
  </h5>
</div>
`;

exports[`GreetingBanner snapshots with size small and morning 1`] = `
<div
  className="d-flex align-items-center justify-content-center p-3.5"
>
  <a
    href="http://localhost:18000/dashboard"
  >
    <Image
      alt="localhost"
      className="d-block"
      src="https://edx-cdn.org/v3/default/logo-white.svg"
      style={
        Object {
          "width": "46px",
        }
      }
    />
  </a>
  <div
    className="greetings-slash-container-small bg-brand-500"
  />
  <h5
    className="text-center text-accent-b"
    role="presentation"
  >
    Good Morning!
  </h5>
</div>
`;
